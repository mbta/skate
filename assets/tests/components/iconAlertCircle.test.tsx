import React from "react"
import renderer from "react-test-renderer"
import IconAlertCircle, {
  AlertIconStyle,
  IconAlertCircleSvgNode,
} from "../../src/components/iconAlertCircle"

test("renders black", () => {
  const tree = renderer
    .create(<IconAlertCircle style={AlertIconStyle.Black} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test("renders greyOnGrey", () => {
  const tree = renderer
    .create(<IconAlertCircle style={AlertIconStyle.GreyOnGrey} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test("renders highlighted", () => {
  const tree = renderer
    .create(<IconAlertCircle style={AlertIconStyle.Highlighted} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test("renders an unwrapped svg node", () => {
  const tree = renderer
    .create(
      <svg>
        <IconAlertCircleSvgNode style={AlertIconStyle.Black} />
      </svg>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
