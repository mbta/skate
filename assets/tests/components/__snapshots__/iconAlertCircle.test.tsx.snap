// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an unwrapped svg node 1`] = `
<svg>
  <g
    className="c-icon-alert-circle--black"
  >
    <circle
      className="c-icon-alert-circle__outline"
      cx="24"
      cy="24"
      r="27"
    />
    <circle
      className="c-icon-alert-circle__fill"
      cx="24"
      cy="24"
      r="22.59"
    />
    <g
      className="c-icon-alert-circle__exclamation-point"
    >
      <path
        d="m20.39 4.42h7.22a1.81 1.81 0 0 1 1.49 2.11l-1.49 23.22a1.73 1.73 0 0 1 -1.49 1.78h-4.24a1.73 1.73 0 0 1 -1.49-1.78l-1.49-23.22c-.07-1.13.61-2.11 1.49-2.11"
      />
      <path
        d="m21.57 34.54h4.87a1.88 1.88 0 0 1 1.82 1.93v5.17a1.89 1.89 0 0 1 -1.82 1.94h-4.87a1.89 1.89 0 0 1 -1.83-1.94v-5.17a1.88 1.88 0 0 1 1.83-1.93"
      />
    </g>
  </g>
</svg>
`;

exports[`renders black 1`] = `
<svg
  className="c-icon-alert-circle"
  viewBox="-3 -3 54 54"
>
  <g
    className="c-icon-alert-circle--black"
  >
    <circle
      className="c-icon-alert-circle__outline"
      cx="24"
      cy="24"
      r="27"
    />
    <circle
      className="c-icon-alert-circle__fill"
      cx="24"
      cy="24"
      r="22.59"
    />
    <g
      className="c-icon-alert-circle__exclamation-point"
    >
      <path
        d="m20.39 4.42h7.22a1.81 1.81 0 0 1 1.49 2.11l-1.49 23.22a1.73 1.73 0 0 1 -1.49 1.78h-4.24a1.73 1.73 0 0 1 -1.49-1.78l-1.49-23.22c-.07-1.13.61-2.11 1.49-2.11"
      />
      <path
        d="m21.57 34.54h4.87a1.88 1.88 0 0 1 1.82 1.93v5.17a1.89 1.89 0 0 1 -1.82 1.94h-4.87a1.89 1.89 0 0 1 -1.83-1.94v-5.17a1.88 1.88 0 0 1 1.83-1.93"
      />
    </g>
  </g>
</svg>
`;

exports[`renders grey 1`] = `
<svg
  className="c-icon-alert-circle"
  viewBox="-3 -3 54 54"
>
  <g
    className="c-icon-alert-circle--grey"
  >
    <circle
      className="c-icon-alert-circle__outline"
      cx="24"
      cy="24"
      r="27"
    />
    <circle
      className="c-icon-alert-circle__fill"
      cx="24"
      cy="24"
      r="22.59"
    />
    <g
      className="c-icon-alert-circle__exclamation-point"
    >
      <path
        d="m20.39 4.42h7.22a1.81 1.81 0 0 1 1.49 2.11l-1.49 23.22a1.73 1.73 0 0 1 -1.49 1.78h-4.24a1.73 1.73 0 0 1 -1.49-1.78l-1.49-23.22c-.07-1.13.61-2.11 1.49-2.11"
      />
      <path
        d="m21.57 34.54h4.87a1.88 1.88 0 0 1 1.82 1.93v5.17a1.89 1.89 0 0 1 -1.82 1.94h-4.87a1.89 1.89 0 0 1 -1.83-1.94v-5.17a1.88 1.88 0 0 1 1.83-1.93"
      />
    </g>
  </g>
</svg>
`;

exports[`renders greyOnGrey 1`] = `
<svg
  className="c-icon-alert-circle"
  viewBox="-3 -3 54 54"
>
  <g
    className="c-icon-alert-circle--grey-on-grey"
  >
    <circle
      className="c-icon-alert-circle__outline"
      cx="24"
      cy="24"
      r="27"
    />
    <circle
      className="c-icon-alert-circle__fill"
      cx="24"
      cy="24"
      r="22.59"
    />
    <g
      className="c-icon-alert-circle__exclamation-point"
    >
      <path
        d="m20.39 4.42h7.22a1.81 1.81 0 0 1 1.49 2.11l-1.49 23.22a1.73 1.73 0 0 1 -1.49 1.78h-4.24a1.73 1.73 0 0 1 -1.49-1.78l-1.49-23.22c-.07-1.13.61-2.11 1.49-2.11"
      />
      <path
        d="m21.57 34.54h4.87a1.88 1.88 0 0 1 1.82 1.93v5.17a1.89 1.89 0 0 1 -1.82 1.94h-4.87a1.89 1.89 0 0 1 -1.83-1.94v-5.17a1.88 1.88 0 0 1 1.83-1.93"
      />
    </g>
  </g>
</svg>
`;

exports[`renders highlighted 1`] = `
<svg
  className="c-icon-alert-circle"
  viewBox="-3 -3 54 54"
>
  <g
    className="c-icon-alert-circle--highlighted"
  >
    <circle
      className="c-icon-alert-circle__outline"
      cx="24"
      cy="24"
      r="27"
    />
    <circle
      className="c-icon-alert-circle__fill"
      cx="24"
      cy="24"
      r="22.59"
    />
    <g
      className="c-icon-alert-circle__exclamation-point"
    >
      <path
        d="m20.39 4.42h7.22a1.81 1.81 0 0 1 1.49 2.11l-1.49 23.22a1.73 1.73 0 0 1 -1.49 1.78h-4.24a1.73 1.73 0 0 1 -1.49-1.78l-1.49-23.22c-.07-1.13.61-2.11 1.49-2.11"
      />
      <path
        d="m21.57 34.54h4.87a1.88 1.88 0 0 1 1.82 1.93v5.17a1.89 1.89 0 0 1 -1.82 1.94h-4.87a1.89 1.89 0 0 1 -1.83-1.94v-5.17a1.88 1.88 0 0 1 1.83-1.93"
      />
    </g>
  </g>
</svg>
`;
